impl/src/main/java/edu/si/trellis/CassandraResource.java

```diff
@@ -133,7 +133,7 @@ public boolean hasAcl() {
     private Stream<Triple> basicContainmentTriples() {
         RDF rdfFactory = TrellisUtils.getInstance();
         Spliterator<Row> rows = bcontainment.execute(getIdentifier()).spliterator();
-        Stream<IRI> contained = StreamSupport.stream(rows, false).map(r -> r.get("contained", IRI.class));
+        Stream<IRI> contained = StreamSupport.stream(rows, false).map(r -> r.get("contained", IRI.class)).distinct();
```

Comment:
Just FYI, you may want to take a look at the API note for [Stream::distinct](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#distinct--). Unless this is absolutely necessary, this call may add quite a bit of overhead, especially for containers with large numbers of child resources.

Reply:
Yes, by the nature of the operation, you have to assume that it could buffer the whole graph, and because of the construction of our RDF, that worst case is the most likely. That's why we're [looking](https://github.com/trellis-ldp/trellis-cassandra/issues/55) at some CQL for this problem.